cache/util/fsutil.go

```diff
@@ -57,16 +57,22 @@ func ReadFile(ctx context.Context, mount snapshot.Mountable, req ReadRequest) ([
 			return errors.WithStack(err)
 		}
 
+		f, err := os.Open(fp)
+		if err != nil {
+			// The filename here is internal to the mount, so we can restore
+			// the request base path for error reporting.
+			// See os.DirFS.Open for details.
+			err.(*os.PathError).Path = req.Filename
```

Comment:
Although this is documented behavior, I don't like this direct type-case. It should be type detection or `errors.As`.